src/site/antora/modules/ROOT/pages/manual/cloud.adoc

```diff
@@ -15,549 +15,267 @@
-    <EventTemplateAdditionalField key="kubernetes.podName" value="${k8s:podName:-}"/>
-    <EventTemplateAdditionalField key="kubernetes.imageId" value="${k8s:imageId:-}"/>
-    <EventTemplateAdditionalField key="kubernetes.imageName" value="${k8s:imageName:-}"/>
-  </JsonTemplateLayout>
-</Socket>
-----
+As explained in <<motivation>>, in a service-oriented architecture, log files are
 
-==== Gelf Template
+* Difficult to maintain – writable volumes must be mounted to the runtime (container, VM, etc.), rotated, and monitored for excessive usage
+* Difficult to use – multiple files need to be manually combined while troubleshooting, no central navigation point
+* Difficult to interoperate – each application needs to be individually configured to produce the same structured log output to enable interleaving of logs from multiple sources while troubleshooting distributed issues
 
-The JsonTemplateLayout can also be used to generate JSON that matches the GELF specification which can format
-the message attribute using a pattern in accordance with the PatternLayout. For example, the following
-template, named EnhancedGelf.json, can be used to generate GELF-compliant data that can be passed to Logstash.
-With this template the message attribute will include the thread id, level, specific ThreadContext attributes,
-the class name, method name, and line number as well as the message. If an exception is included it will also
-be included with newlines. This format follows very closely what you would see in a typical log file on disk
-using the PatternLayout but has the additional advantage of including the attributes as separate fields that
-can be queried.
+In short, *we don't recommend writing logs to files*.
 
-[source,json]
-----
-{
-    "version": "1.1",
-    "host": "${hostName}",
-    "short_message": {
-        "$resolver": "message",
-        "stringified": true
-    },
-    "full_message": {
-        "$resolver": "message",
-        "pattern": "[%t] %-5p %X{requestId, sessionId, loginId, userId, ipAddress, corpAcctNumber} %C{1.}.%M:%L - %m",
-        "stringified": true
-    },
-    "timestamp": {
-        "$resolver": "timestamp",
-        "epoch": {
-            "unit": "secs"
-        }
-    },
-    "level": {
-        "$resolver": "level",
-        "field": "severity",
-        "severity": {
-            "field": "code"
-        }
-    },
-    "_logger": {
-        "$resolver": "logger",
-        "field": "name"
-    },
-    "_thread": {
-        "$resolver": "thread",
-        "field": "name"
-    },
-    "_mdc": {
-        "$resolver": "mdc",
-        "flatten": {
-            "prefix": "_"
-        },
-        "stringified": true
-    }
-}
-----
+A common excuse for writing logs to files is to use them as a buffer prior to ingesting them either to the logging system proxy or to the log storage system.
+Even in this case, most of the time it is sign of bad architecture design, and costs extra resources and efficiency degradation.
```

Comment:
Isn't this what proxies are doing?

Flume is out of shape, but it has such a functionality (as far as I understand it is the only functionality worth splitting).

Reply:
What I mean is that the practice of buffering logs in logs files might still be valid.
However the logic that does that probably should not be in the application itself, but in the sidecar service that forwards the logs from the application.

Reply:
Removed the paragraph in fed8ac44975b212f60b070c3a6aef34fe0023d6b.